### ISSUE TITLE
Not much bounds checking is done when moving selection window

### ISSUE THREAD

**Issue by justinhj** [justinhj/rust-mandelbrot#2]
When moving the selection window there is some but not enough checking for bounds. Specifically when you reach one or more edges of the screen the selection window should not move in that direction.

**Comment by tusharhqq** [comment]
is there a way to know the screen width of the device on which we are viewing the image. Cause for the bounds I was thinking of creating something like this
// Check if the new corners exceed the bounds of the screen
    let max_re = screen_width;
    let min_re = 0.0;
    let max_im = screen_height;
    let min_im = 0.0;

am I going in right direction for this problem or I am completely wrong

**Comment by tusharhqq** [comment]
can u tell me the direction in which I need to think for this problem

**Comment by justinhj** [comment]
Yeah you're on the right lines. There are functions such as move_selection_right, move_selection_down and so on. You need to calculate whether the thing is going out of bounds. Sorry it's been a while since I looked at the code, but there are essentially two coordinate systems in play, the terminal coords and the window coords. You may need to add some print! instructions to the various variables and print them out while moving things around and see what they do. The idea is to figure out the bounds and keep the user within them.

**Comment by justinhj** [comment]
Fixed in latest
